jbpm-test-coverage/src/test/java/org/jbpm/test/functional/event/BoundaryEventWithOutputSetPersistenceTest.java

```diff
@@ -0,0 +1,415 @@
+        // Register Human Task Handler
+        ksession.getWorkItemManager().registerWorkItemHandler(WORK_ITEM_HANDLER_TASK, humanTaskMockHandler);
+        ksession.getWorkItemManager().registerWorkItemHandler(WORK_ITEM_HANDLER_DATA, dataTaskMockHandler);
+        // Start the process 
+        ProcessInstance process = ksession.startProcess(PROCESS_CONDITIONAL_NAME);
+        long processId = process.getId();
+        assertTrue("process id not saved", processId > 0);
+        
+        // The process is in the Human Task waiting for its completion
+        int processState = process.getState();
+        assertEquals("Expected process state to be " + processStateName[ProcessInstance.STATE_ACTIVE] + " not "
+                + processStateName[processState], ProcessInstance.STATE_ACTIVE, processState);
+
+        ksession.signalEvent("RuleFlowStateEvent-" + PROCESS_CONDITIONAL_NAME + "-4-_2" , new org.jbpm.test.entity.Person(1L, "john"));
+       
+        completeWork(ksession, humanTaskMockHandler);
+
+        // The process completes
+        process = ksession.getProcessInstance(process.getId());
+        assertNull("Expected process to have been completed and removed", process);
+        assertTrue(dataTaskMockHandler.getInputParameter("nodeInstance") instanceof HumanTaskNodeInstance);
+        assertTrue(dataTaskMockHandler.getInputParameter("event") instanceof Person);
+        assertEquals(dataTaskMockHandler.getInputParameter("signal"), "RuleFlowStateEvent-BoundaryConditionalEventProcess-4-_2");
+    }
+    
+    private void completeWork(KieSession ksession, HumanTaskMockHandler humanTaskMockHandler) {
+        assertTrue("The work item task handler does not have a work item!", humanTaskMockHandler.workItem != null);
+        long workItemId = humanTaskMockHandler.workItem.getId();
+        assertTrue("work item id not saved", workItemId > 0);
+        
+        // The Human Task is completed
+        Map<String, Object> results = new HashMap<String, Object>();
+        try {
+            ksession.getWorkItemManager().completeWorkItem(workItemId, results);
+        } catch (Exception e) {
+            logger.warn("Work item could not be completed!");
+            e.printStackTrace();
+            fail(e.getClass().getSimpleName() + " thrown when completing work item: " + e.getMessage());
+        }
+    }
+
+    private void sleepAndVerifyTimerRuns()  { 
+        try {
+            int sleep = 3000;
+            logger.debug("Sleeping {} seconds", sleep / 1000);
+            Thread.sleep(sleep);
+            logger.debug("Awake!");
+        } catch(InterruptedException e) {
+            logger.error("System was interrupted");
+            fail("Junit was interrupted");
+        }
+
+    }
+
+
+
+    private static class HumanTaskMockHandler implements WorkItemHandler {
+
+        private org.kie.api.runtime.process.WorkItemManager workItemManager;
+        private org.kie.api.runtime.process.WorkItem workItem;
```

Comment:
we don't need to use fully qualified class name for them

Reply:
removed